bulk_lookup/csv.py

```diff
@@ -0,0 +1,59 @@
+import os
+import mmap
+import pyexcel
+from pyexcel._compact import zip_longest
+import defusedxml
+
+defusedxml.defuse_stdlib()
+
+
+class mmmap(mmap.mmap):
+    """A mmap that doesn't have a read function. This is so that
+    pyexcel-io doesn't pass this through as a stream (where it
+    won't work. https://github.com/pyexcel/pyexcel-io/issues/34 )"""
+    def __getattribute__(self, name):
+        if name == 'read':
+            raise AttributeError
+        return super(mmmap, self).__getattribute__(name)
+
+
+class PyExcelReader(object):
+    # Set up an iterator for reading in a CSV/XLS/XLSX/ODS file
+    def __init__(self, file):
```

Comment:
I'd probably rename this parameter `file_field`, since I assumed this was a normal file-like object and then got puzzled about where the `.open()` method came from. Also then it's not shadowing the `file` constructor.